### PR TITLE
Update deps

### LINKED ISSUE #747: Bump Tor version for 2.0 release
We should fetch the Tor/obfs4proxy binaries from TBB 8.0.0 probably, prior to releasing 2.0.

### PR DESCRIPTION
**Pull request by micahflee**
For Windows and Mac, update to python 3.7.0 and Qt 5.11.1, and also update tor binaries from Tor Browser 8.0.

This will resolve #747.

### CODE REVIEW

File: install/get-tor-osx.py [hunk 1]
```diff
@@ -37,7 +37,7 @@
 def main():
     dmg_url = 'https://archive.torproject.org/tor-package-archive/torbrowser/7.5.5/TorBrowser-7.5.5-osx64_en-US.dmg'
     dmg_filename = 'TorBrowser-7.5.5-osx64_en-US.dmg'
```
Comment (mig5) [inline install/get-tor-osx.py]: @micahflee the version of the .dmg here is still 7.5.5, was it meant to be updated to 8.0 like in `get-tor-windows.py`?

Comment (micahflee) [reply install/get-tor-osx.py]: Oh hah good review. I tested the actual packaging in Windows, but didn't test the actual packaging in OSX

**Comment by mig5** [comment]
Dang - it looks like maybe something's changed inside the dmg as to the Tor location.

```
Downloading https://archive.torproject.org/tor-package-archive/torbrowser/8.0/TorBrowser-8.0-osx64_en-US.dmg
Checksumming whole disk (unknown partition : 0)…
....................................................................................................................................................................................
  whole disk (unknown partition : 0): verified   CRC32 $55C04439
verified   CRC32 $8728B4C5
/dev/disk4          	                               	/Volumes/Tor Browser
Traceback (most recent call last):
  File "/Users/miguel/git/onionshare/install/get-tor-osx.py", line 101, in <module>
    main()
  File "/Users/miguel/git/onionshare/install/get-tor-osx.py", line 82, in main
    tor_script = open(os.path.join(dmg_tor_path, 'Resources', 'TorBrowser', 'Tor', 'tor'), 'r').read()
FileNotFoundError: [Errno 2] No such file or directory: '/Volumes/Tor Browser/TorBrowser.app/Contents/Resources/TorBrowser/Tor/tor'
```

I'll try and debug and might take the liberty of pushing to your branch to fix if possible, just to move things along.

**Comment by mig5** [comment]
Something very weird is going on here.

I have a fix for the get-tor-osx.py and dealing with the .dmg, but  the install/requirements.txt in this branch is somehow now missing all the new dependencies for v3 onions, as well as 'requests' which was previously there... https://github.com/micahflee/onionshare/pull/731/files#diff-f3fdb6f7f827f65b52348d8f0b72e0bd

**Comment by mig5** [comment]
In fact those changes are missing from install/requirements.txt in *develop* branch - so how did those get demolished when we merged receiver-mode-gui into develop??

**Comment by mig5** [comment]
So somehow, when develop got merged into receiver-mode-gui earlier, it undid several changes that were pending for v3 support :/

https://github.com/micahflee/onionshare/commit/6efc281fbbb5761a13e7b821cecc894d0aee2fbc

**Comment by mig5** [comment]
I fixed the OS X issues, but then encountered new dependency issues with Windows.

The Crypto python dependency (introduced for v3 support) fails to build, citing 'error: Microsoft Visual C++ 14.0 is required. Get it with "Microsoft Visual C++ Build Tools": http://landinghub.visualstudio.com/visual-cpp-build-tools'

That page 404s, but I found my way to https://visualstudio.microsoft.com/thank-you-downloading-visual-studio/?sku=BuildTools&rel=15#/. (I downloaded vs_buildtools_2119740225.1537250142.exe)

Then I promptly ran out of disk space in my VM for the Qt install (38GB+ required, ouch)...

Can we give up on supporting Windows yet? :) :)

Maybe in our opposing timezones you might get a bit further with the Windows side of things. The rest is looking good and my OS X install worked.

**Comment by mig5** [comment]
Even after I installed the visual studio tools I got horrible 'syntax error: missing '{'' when I retried my `pip3 install -r install\requirements-windows.txt`...

**Comment by micahflee** [comment]
Oh damn. When I merged develop into receiver-mode-gui, I dealt with an enormous merge, and I possibly messed up the requirements.txt file then. When I first made this PR, I tested installing the dependencies and packaging in Windows, and I didn't run into the Visual C++ thing. Maybe it's because I already have the Windows SDK installed in my VM, and that's what was needed? But ok, I'll take another look and try to finish this PR today.

**Comment by mig5** [comment]
What I wonder is due to that merge, maybe your testing of installing the dependencies simply lacked the *new* dependencies and that's how you didn't run into the compile issue? Since the compile issue is against pycrypto which is a new dependency.

Apparently a solution for my pycrypto compile errors on Windows, is to use pycryptodome instead of pycrypto:

https://stackoverflow.com/questions/42339631/python-3-how-to-set-visual-studio-c-2015-compiler

https://pypi.org/project/pycryptodome/

I just replaced pycrypto with pycryptodome==3.6.6 in install/requirements-windows.txt and it worked, I can start OnionShare! So maybe we just use that as a dependency on windows instead?

Do you want to do a fresh clone of this branch in your Windows, try installing, and if you encounter issues with pycrypto's setup too, switch to pycryptodome? If you don't, maybe you're just lucky and we can move on :)

File: install/get-tor-osx.py [hunk 2]
```diff
@@ -88,7 +88,7 @@ def main():
     shutil.copyfile(os.path.join(dmg_tor_path, 'Resources', 'TorBrowser', 'Tor', 'geoip6'), os.path.join(dist_path, 'Resources', 'Tor', 'geoip6'))
     os.chmod(os.path.join(dist_path, 'Resources', 'Tor', 'tor'), 0o755)
     shutil.copyfile(os.path.join(dmg_tor_path, 'MacOS', 'Tor', 'tor.real'), os.path.join(dist_path, 'MacOS', 'Tor', 'tor.real'))
-    shutil.copyfile(os.path.join(dmg_tor_path, 'MacOS', 'Tor', 'libevent-2.0.5.dylib'), os.path.join(dist_path, 'MacOS', 'Tor', 'libevent-2.0.5.dylib'))
+    shutil.copyfile(os.path.join(dmg_tor_path, 'MacOS', 'Tor', 'libevent-2.1.6.dylib'), os.path.join(dist_path, 'MacOS', 'Tor', 'libevent-2.1.6.dylib'))
```
[the remark this answers is not in the document]
Comment (micahflee) [reply install/get-tor-osx.py]: Thanks! I'll try Mac and Windows packaging

**Comment by micahflee** [comment]
I'm going to clean up the pip dependencies, and see how possibly it is to use a single requirements.txt file for both Mac and Windows.

**Comment by mig5** [comment]
Sounds good. And if pycryptodome is more successful on Windows (or was it 'just me'), and seems compatible with pycrypto, maybe it can be used on MacOS too..

**Comment by mig5** [comment]
Hooray for that .nsi change!!!

I'm just doing a quick Windows test now - all good on macOS. Then I'll merge.

File: install/requirements.txt [hunk 3]
```diff
@@ -10,12 +13,15 @@ Jinja2==2.10
...
 pefile==2018.8.8
+pycparser==2.18
+pycryptodome==3.6.6
 PyInstaller==3.4
+PyNaCl==1.2.1
 PyQt5==5.11.2
 PyQt5-sip==4.19.12
 PySocks==1.6.8
 requests==2.19.1
-sip==4.19.8
```
Comment (mig5) [inline install/requirements.txt]: Got a warning when packaging on my Windows 'WARNING: hidden import 'sip' not found'. Not sure if important

**Pull request merged**